### PR TITLE
Correct PyObjectArenaAllocator to match description in python docs

### PR DESCRIPTION
**Pull request by ryancinsight**
Pyobjectarenaallocator free should be [free(void *ctx, size_t size, void *ptr)]

according to https://docs.python.org/3/c-api/memory.html

#257 
@indygreg

### CODE REVIEW

**Review by indygreg — APPROVED** [review]
This change looks good to me!

But I'm not a project maintainer so I can't merge it for you. I'm also unsure if this should be documented in release notes anywhere. I'll let an actual maintainer chime in :)

**Comment by markbt** [comment]
Thanks!
It looks like this definition isn't used elsewhere in rust-cpython, so it would only affect anyone accessing it directly through `python3-sys`.  I'm guessing not many people have done that.

**Comment by markbt** [comment]
Oh, I double-checked the code, and it looks like it's actually the **Python docs** that are wrong: [see the definition in CPython itself](https://github.com/python/cpython/blob/master/Include/cpython/objimpl.h#L72).

**Comment by ryancinsight** [comment]
Yeah, we are trying to customize the Python allocators used in portable Python packages. Ahh, I should have tracked it down further, I’ll get in touch with them to figure this out.

**Comment by indygreg** [comment]
> Oh, I double-checked the code, and it looks like it's actually the **Python docs** that are wrong: [see the definition in CPython itself](https://github.com/python/cpython/blob/master/Include/cpython/objimpl.h#L72).

CC @vstinner

**Comment by markbt** [comment]
Going to close this PR as it's the Python docs that are wrong, and I don't want it to get accidentally merged.

**Comment by vstinner** [comment]
> it's the Python docs that are wrong

Oops. I proposed to https://github.com/python/cpython/pull/24795 to fix.

The PEP is as old as the PEP 445 implementation: https://github.com/python/cpython/commit/0507bf56f0b1064a179464eb3a9d524fd6dad815

**Pull request closed without merging**
